apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/db/DbMultiDomainAcsStore.scala

```diff
@@ -229,11 +229,31 @@ final class DbMultiDomainAcsStore[TXE](
       .value
   }
 
-  def hasArchived(ids: Seq[ContractId[?]])(implicit
-      traceContext: TraceContext
-  ): Future[Boolean] =
-    // TODO(#838): implement this as a single DB query
-    Future.sequence(ids.map(lookupContractStateById)).map(_.exists(_.isEmpty))
+  def containsArchived(ids: Seq[ContractId[?]])(implicit
+                                                traceContext: TraceContext
+  ): Future[Boolean] = waitUntilAcsIngested {
+    if (ids.isEmpty) Future.successful(true)
+    else {
+      val contractIds = inClause(ids)
+      val expectedCount = ids.size
+      storage.query(
+        (sql"""
+        select count(1) from (
+           select acs.contract_id
```

Comment:
```suggestion
           select count(1)
```

Does this not work? I don't understand why you need a nested select.

Reply:
yeah that will work too